tests/common/dualtor/data_plane_utils.py

```diff
@@ -144,13 +146,24 @@ def run_test(duthosts, activehost, ptfhost, ptfadapter, action,
     send_and_sniff.set_error_handler(lambda *args, **kargs: io_ready.set())
 
     send_and_sniff.start()
+    io_ready.wait()
     if action:
         # do not perform the provided action until
         # IO threads (sender and sniffer) are ready
-        io_ready.wait()
         logger.info("Sender and sniffer threads started, ready to execute the "\
             "callback action")
         action()
+    # do not time-wait the test, if early start is not requested (when stop_after=0)
+    if stop_after:
+        start_time = datetime.datetime.now()
+        # Wait until timeout happens OR the IO test completes
+        while ((datetime.datetime.now() - start_time).seconds < stop_after) and\
```

Comment:
Could we re-use this `wait_until` function here? https://github.com/Azure/sonic-mgmt/blob/a344b50bc76f9d371a52c651658e8e2d2b3d5523/tests/common/utilities.py#L35

Reply:
Good idea! Updated now with `wait_until`